Dockerfile

```diff
@@ -3,7 +3,7 @@
 # Copyright 2020-2023 the Pinniped contributors. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-ARG BUILD_IMAGE=golang:1.21.4@sha256:9baee0edab4139ae9b108fffabb8e2e98a67f0b259fd25283c2a084bd74fea0d
```

Comment:
There is actually a new 1.21.5 release, but the automation is accidentally getting s `latest`. So we want this bump, but we don't want to call it `latest` and we want this sha:

```
❯ docker run --rm gcr.io/go-containerregistry/crane digest golang:1.21.5 --platform all
sha256:58e14a93348a3515c2becc54ebd35302128225169d166b7c6802451ab336c907
```

I'll disable the `latest` version in the Concourse resource and rerun this CI job, and it should work.

Reply:
I think this is a bug in the Concourse `registry-image` resource. According to the [docs](https://github.com/concourse/registry-image-resource) it should only be returning semver version tags, not `latest`.